### PR TITLE
HIVE-24791: Backward compatibility issue in _dumpmetadata

### PR DESCRIPTION
**Pull request by ArkoSharma**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ private void loadDumpFromFile() throws SemanticException {
           lineContents[2].equals(Utilities.nullStringOutput) ? null : Long.valueOf(lineContents[2]),
           lineContents[3].equals(Utilities.nullStringOutput) ? null : new Path(lineContents[3]),
           lineContents[4].equals(Utilities.nullStringOutput) ? null : Long.valueOf(lineContents[4]),
-          Boolean.valueOf(lineContents[6]));
+          (lineContents.length < 8 || lineContents[6].equals(Utilities.nullStringOutput)) ?
```
Comment (aasha) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:134]: check for length < 7

Comment (aasha) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:134]: Add a test

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java [hunk 2]
```diff
@@ -203,7 +204,7 @@ private void initializeIfNot() throws SemanticException {
     }
   }
 
-  private List<String> prepareReplScopeValues() {
+  public List<String> prepareReplScopeValues() {
```
Comment (aasha) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:207]: why is this needed?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
 public class DumpMetaData {
   // wrapper class for reading and writing metadata about a dump
   // responsible for _dumpmetadata files
-  private static final String DUMP_METADATA = "_dumpmetadata";
+  public static final String DUMP_METADATA = "_dumpmetadata";
```
Comment (aasha) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:41]: why is this needed?

Comment (ArkoSharma) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:41]: The test for this patch (testDumpMetadataBackwardCompatibility) needs to probe on the dumpmetadata file.
The test works by doing a dump, then deleting newly introduced fields from the dumpmetadata file and then doing a load. This change was made to provide access to the dmd-file name to the test.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java (hunk 2 above)
Comment (ArkoSharma) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/DumpMetaData.java:207]: Done to give access to this function from the unit-test (testDumpMetadataBackwardCompatibility). This function is required to create the repl-scope values during dump. So, during the test, while re-creating the dmd file in the same manner, this function is used.
